ConnectorTeklaStructures/ConnectorTeklaStructuresShared/MainForm.cs

```diff
@@ -24,41 +24,89 @@ public partial class MainForm : PluginFormBase
   const int GWL_HWNDPARENT = -8;
 
   private static Avalonia.Application AvaloniaApp { get; set; }
+
   public Model Model { get; private set; }
+
   public static Window MainWindow { get; private set; }
+
   public static ConnectorBindingsTeklaStructures Bindings { get; set; }
 
+  /// <summary>
+  /// Initializes a new instance of the MainForm class.
+  /// </summary>
   public MainForm()
+  {
+    InitializeComponents();
+    SetupApplication();
+  }
+
+  /// <summary>
+  /// Initializes the components of the MainForm.
+  /// </summary>
+  private void InitializeComponents()
   {
     Load += MainForm_Load;
-    if (MainWindow == null)
+
+    if (MainWindow != null)
     {
-      // Link to model.
-      Model = new Model();
-      Bindings = new ConnectorBindingsTeklaStructures(Model);
+      return;
+    }
 
+    Model = new Model();
+    Bindings = new ConnectorBindingsTeklaStructures(Model);
+  }
+
+  /// <summary>
+  /// Sets up the application, including event subscriptions, UI build, and model bindings.
+  /// </summary>
+  private static void SetupApplication()
+  {
+    if (MainWindow == null)
+    {
       try
       {
-        AppDomain.CurrentDomain.AssemblyResolve += new ResolveEventHandler(OnAssemblyResolve);
-
-        Setup.Init(Bindings.GetHostAppNameVersion(), Bindings.GetHostAppName());
-        BuildAvaloniaApp().Start(AppMain, null);
-        var viewModel = new MainViewModel(Bindings);
-        MainWindow = new DesktopUI2.Views.MainWindow { DataContext = viewModel };
-
-        Bindings.OpenTeklaStructures();
+        SubscribeToEvents();
+        BuildAndShowMainWindow();
+        SetTeklaAsOwner();
       }
-      catch (Exception ex)
+      catch (Exception ex) when (!ex.IsFatal())
```

Comment:
This is the example I was referring to - most of the DUI c# connectors have approximately the same general Exception catch and some previously swallowed. Probably a candidate for coalescing.

Reply:
Yes we have a ticket to handle the top-level catch block in the backlog, but for now it's probably worth catching Exception here and disabling the warning, unless you're saying something that calls this WILL catch(Exception)

Reply:
No catch exists above this. A failure to create the plugin window shouldn't disable the host application.